eternalcore-core/src/main/java/com/eternalcode/core/feature/warp/WarpInventory.java

```diff
@@ -70,19 +74,29 @@ public class WarpInventory {
         this.warpTeleportService = warpTeleportService;
         this.configurationManager = configurationManager;
         this.config = config;
+        this.scheduler = scheduler;
     }
 
     public void openInventory(Player player) {
-        this.languageService.getLanguage(player.getUniqueId()).whenComplete((language, throwable) -> {
-            if (language == null) {
-                language = Language.DEFAULT;
-            }
+        CompletableFuture<Gui> guiFuture = this.createInventoryAsync(player);
 
-            this.createInventory(player, language)
-                .open(player);
+        guiFuture.thenAccept(gui -> this.scheduler.run(() -> gui.open(player))).exceptionally(throwable -> {
+            CompletableFutureUtil.handleCaughtException(throwable);
+            return null;
         });
     }
 
+    private CompletableFuture<Gui> createInventoryAsync(Player player) {
+        return this.languageService.getLanguage(player.getUniqueId())
+            .thenApply(language -> {
+                if (language == null) {
+                    language = Language.DEFAULT;
+                }
+                return this.createInventory(player, language);
```

Comment:
I suggest adding a method to `languageService#getOrCreateLanguage` that retrieves a language or creates a default one if it is null. This way, the code would be
```suggestion
        return this.languageService.getOrCreateLanguage(player.getUniqueId())
            .thenApply(language -> {
                return this.createInventory(player, language);
```

Reply:
here we have notnull variable, we can remove this if condition